harbor/tests/conftest.py

```diff
@@ -0,0 +1,169 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import os
+import time
+
+import pytest
+import requests
+from mock import MagicMock, patch
+
+from datadog_checks.dev import LazyFunction, docker_run
+from datadog_checks.dev.conditions import CheckDockerLogs
+from datadog_checks.harbor import HarborCheck
+from datadog_checks.harbor.api import HarborAPI
+from datadog_checks.harbor.common import (
+    CHARTREPO_HEALTH_URL,
+    HEALTH_URL,
+    LOGIN_PRE_1_7_URL,
+    LOGIN_URL,
+    PING_URL,
+    PROJECTS_URL,
+    REGISTRIES_PING_PRE_1_8_URL,
+    REGISTRIES_PING_URL,
+    REGISTRIES_PRE_1_8_URL,
+    REGISTRIES_URL,
+    SYSTEM_INFO_URL,
+    VOLUME_INFO_URL,
+)
+
+from .common import (
+    ADMIN_INSTANCE,
+    CHARTREPO_HEALTH_FIXTURE,
+    HARBOR_VERSION,
+    HEALTH_FIXTURE,
+    HERE,
+    INSTANCE,
+    PROJECTS_FIXTURE,
+    REGISTRIES_FIXTURE,
+    REGISTRIES_PRE_1_8_FIXTURE,
+    SYSTEM_INFO_FIXTURE,
+    URL,
+    USERS_URL,
+    VERSION_1_6,
+    VERSION_1_8,
+    VOLUME_INFO_FIXTURE,
+)
+
+UNTRACKED_FILES = [
+    os.path.join('common', 'config', 'core', 'certificates'),
+    os.path.join('common', 'config', 'custom-ca-bundle.crt'),
+    os.path.join('common', 'config', 'ui', 'certificates'),
+    os.path.join('data', 'ca_download'),
+    os.path.join('data', 'chart_storage'),
+    os.path.join('data', 'config'),
+    os.path.join('data', 'job_logs'),
+    os.path.join('data', 'psc'),
+    os.path.join('data', 'redis'),
+    os.path.join('data', 'registry'),
+]
+
+
+@pytest.fixture(scope='session')
+def dd_environment(instance):
+    compose_file = get_docker_compose_file()
+    conditions = [
+        CheckDockerLogs(compose_file, "http server Running on", wait=3),
+        lambda: time.sleep(2),
+        CreateSimpleUser(),
+    ]
+    with docker_run(compose_file, conditions=conditions):
+        yield instance
+
+
+class CreateSimpleUser(LazyFunction):
+    def __call__(self, *args, **kwargs):
+        with requests.session() as session:
+            harbor_api = HarborAPI(URL, session)
+            harbor_api.authenticate('admin', 'Harbor12345')
+            harbor_api._make_post_request(
+                USERS_URL,
+                json={
+                    "username": "NotAnAdmin",
+                    "email": "NotAnAdmin@goharbor.io",
+                    "password": "Str0ngPassw0rd",
+                    "realname": "Not An Admin",
+                },
+            )
+
+
+@pytest.fixture(scope='session')
+def instance():
+    return INSTANCE.copy()
+
+
+@pytest.fixture(scope='session')
+def admin_instance():
+    return ADMIN_INSTANCE.copy()
+
+
+@pytest.fixture
+def harbor_check(admin_instance):
+    check = HarborCheck('harbor', {}, [admin_instance])
+    check.log = MagicMock()
+    check.gauge = MagicMock()
```

Comment:
You shouldn't do that, and use the aggregator instead, unless I miss something.

Reply:
👍